### PR TITLE
Typedoc integration

### PR DESCRIPTION
**Pull request by rahul1**
This PR integrates our Typedoc SDK docs into our docusaurus documentation using the plugin [docusaurus-plugin-typedoc](https://github.com/tgreyuk/typedoc-plugin-markdown/tree/master/packages/docusaurus-plugin-typedoc). The major changes are:

1. A new "SDK" section on the sidebar to house all the typedoc documentation
2. Updates to the typedoc configurations to hide all the private members
3. Some re-ordering/re-naming of existing sidebar elements to create a more logical flow

### CODE REVIEW

File: packages/docs/docs/app/_category_.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "label": "App",
-  "position": 4
+  "position": 5.1
```
Comment (codyebberson) [inline packages/docs/docs/app/_category_.json:3]: I totally understand this, but it also feels a little stinky.  I want to research how other big docusaurus sites manage this.

File: packages/docs/docs/home.md [hunk 2]
```diff
@@ -8,15 +8,15 @@ import HomepageCallout from '@site/src/components/HomepageCallout'
...
+    <HomepageCallout title="Get Started" body="Write your first medical application in 5 minutes" linkText="Read More" linkRef="/intro" />
     <HomepageCallout title="API Docs" body="Reference documentation about Medplum's client API for reading and writing healthcare data" linkText="Read More" linkRef="/api/overview" />
     <HomepageCallout title="Use Cases" body="See how to apply Medplum against your healthcare problem" linkText="Read More" linkRef="#" />
     <HomepageCallout title="Basic Concepts" body="Learn the basic concepts behind Medplum and the FHIR standard for healthcare data" linkText="Read More" linkRef="#" />
 </section>
 
 ## What is Medplum?
 
-Medplim is a developer platform that enables **flexible and rapid development** of healthcare apps. In consists of 3 main components
+Medplum is a developer platform that enables **flexible and rapid development** of healthcare apps. In consists of 3 main components
```
Comment (codyebberson) [inline packages/docs/docs/home.md:19]: 😬

File: packages/docs/docs/self_hosting/_category_.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Self-Hosting",
```
Comment (codyebberson) [inline packages/docs/docs/self_hosting/_category_.json:2]: Nice 💯

File: packages/docs/package.json [hunk 4]
```diff
@@ -37,6 +37,9 @@
   "devDependencies": {
     "@docusaurus/module-type-aliases": "2.0.0-beta.20",
     "@tsconfig/docusaurus": "1.0.5",
+    "docusaurus-plugin-typedoc": "^0.17.5",
```
Comment (codyebberson) [inline packages/docs/package.json:40]: Minor, but I've been trying to maintain pinned versions of everything to keep cross-package versions more sane.  Prefer to drop the "^"

**Review by codyebberson — COMMENTED** [review]
Overall lgtm 👍

File: packages/docs/docs/app/_category_.json (hunk 1 above)
Comment (rahul1) [reply packages/docs/docs/app/_category_.json:3]: The "real" way to do this is to configure a sidebar manually in the `docusaurus.config.json` file. Would like to do that in a separate re-factoring pass

File: packages/docs/package.json (hunk 4 above)
Comment (rahul1) [reply packages/docs/package.json:40]: yeah that's fine. This was automatic

File: packages/docs/docs/app/_category_.json (hunk 1 above)
Comment (rahul1) [reply packages/docs/docs/app/_category_.json:3]: For example, take a look at supabase: https://github.com/supabase/supabase/blob/master/web/sidebars.js

**Review by codyebberson — APPROVED** [review]
👍

**Pull request merged**